ADR/0033-provisioning-test-resources.md

```diff
@@ -0,0 +1,279 @@
+# 33. Provisioning Test Resources
+
+Date: 2024-02-14
+
+## Status
+
+Accepted
+
+Supersedes:
+
+- [ADR 08. Environment Provisioning](0008-environment-provisioning.html)
+- Environment provisioning parts of [ADR 32. Decoupling Deployment](0032-decoupling-deployment.html)
+
+## Context
+
+This decision clarifies how integration test environments will be dynamically provisioned. In prior
+decisions it was believed that [Dynamic Resource Allocation] (DRA) would graduate out of OpenShift
+TechPreview on a timeline suitable for this project. This is no longer the case and, as such, a new
+approach to requesting compute resources for test pipelines is needed. DRA is still expected to
+become the kubernetes-native approach to managing dynamically provisioned resources across pods.
+Therefore, any interim solution should intentionally avoid introducing new barriers that might
+prevent the adoption of DRA some day.
+
+The problem of provisioning test resources/environments can be broken down into a few questions:
+
+1. How can resources (OpenShift clusters, to start) be provisioned, efficiently, without exposing
+   cloud/infra credentials to the end user?
+1. How does the user declare ephemeral resources and inject their resulting secrets into
+   `TaskRuns`/`Pods`?
+1. How can ephemeral resources be shared across `TaskRuns`/`Pods`?
+
+## Decision
+
+### Cluster Provisioning
+
+[Hive] will be used to provision OpenShift clusters. It's widely used in OpenShift CI testing,
+supports hibernated cluster pools for quick (5-7m), cost-efficient, allocation and is maintained
+and distributed by Red Hat on OperatorHub. [Cluster API] is an intriguing option for provisioning
+Kubernetes and OpenShift clusters but expected to remain in OpenShift TechPreview throughout 2024.
+It also recommends using a dedicated management cluster separate from application workloads which
+creates additional infrastructure complexity.
+
+Hive `ClusterPools` and `ClusterImageSets` will be maintained by Konflux admins (possibly with the
```

Comment:
IMHO, what I like about the openshift-ci platform approach is it lets me do a "bring your own cluster pool config".  Some teams might have different cluster configurations than others and may need different flavor sizes. 

What is not clear is that I can't tell if that is still possible and I just need to had these configs over to a Konflux Admin to configure which is fine or if Konflux is predefining and there isn't many options?